bike_kollective/lib/data/provider/database.dart

```diff
@@ -19,7 +19,8 @@ final databaseProvider = Provider<BKDB>((ref) {
 // Interface for Bike Kollective database (implementations are below)
 abstract class BKDB {
   // User CRUD operations
-  Future<UserModel> addUser(String uid) ;
+  Future<UserModel> addUser(String uid, String email) ;
+  Future<UserModel> addUserEmail(String email, String password);
```

Comment:
Combine with addUser() and make the password optional? I'm not really following this case well. Also, again, we shouldn't be storing raw passwords in the database.